### PR TITLE
[005] Add CDC example

### PR DESCRIPTION
**Pull request by jbartok**
Checklist
- [x] Tags Set
- [x] Milestone Set

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/233/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/233/jet-oss-pr-builder-docs-no-build-233.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: examples/README.md
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/234/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/234/jet-oss-pr-builder-docs-no-build-234.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: examples/README.md
</details>
</pre>

**Comment by jbartok** [comment]
verify

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/235/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/235/jet-oss-pr-builder-docs-no-build-235.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: examples/README.md
</details>
</pre>

**Comment by jbartok** [comment]
verify

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder-docs-no-build` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/236/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder-docs-no-build/236/jet-oss-pr-builder-docs-no-build-236.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
There are unexpected files.
Unexpected file: examples/README.md
</details>
</pre>

**Comment by jbartok** [comment]
verify

**Comment by jbartok** [comment]
verify

File: examples/cdc/src/main/java/com/hazelcast/jet/examples/cdc/Cache.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(source)
+                .withoutTimestamps()
+                .peek()
+                .writeTo(CdcSinks.map("customers",
+                        r -> r.key().toMap().get("id"),
+                        r -> r.value().toObject(Customer.class).toString()));
+
+        JobConfig cfg = new JobConfig().setName("mysql-monitor");
+        Jet.bootstrappedInstance().newJob(pipeline, cfg);
```
Comment (cangencer) [inline examples/cdc/src/main/java/com/hazelcast/jet/examples/cdc/Cache.java:70]: should there a be a .`join()` call here?

Comment (jbartok) [reply examples/cdc/src/main/java/com/hazelcast/jet/examples/cdc/Cache.java:70]: Hmm... I guess if you run it stand-alone, then yes, you want to stick around, not like when you submit it to the cluster... So I guess, yes, safer to add the join. Will do.

File: examples/cdc/src/main/java/com/hazelcast/jet/examples/cdc/Cache.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                .setTableWhitelist("inventory.customers")
+                .build();
+
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(source)
+                .withoutTimestamps()
+                .peek()
+                .writeTo(CdcSinks.map("customers",
+                        r -> r.key().toMap().get("id"),
+                        r -> r.value().toObject(Customer.class).toString()));
```
Comment (cangencer) [inline examples/cdc/src/main/java/com/hazelcast/jet/examples/cdc/Cache.java:68]: very minor but the `);` should be on next line

**Pull request merged**